File: gateway/src/core/card-payment/CardPayment.ts

export enum PaymentType {
  CREDIT_CARD ='CREDIT_CARD',
  DEBIT_CARD = 'DEBIT_CARD'
}

export type CardPayment = {
  clientId: string
  paymentType: PaymentType
  paymentDate: string
  amount: number
  paymentInfo: {
    maskedNumber: string
  }
}

File: gateway-go/src/core/card_payment/card_payment.go
package card_payment

type CardPayment struct {
	Id          string          `json:"id"`
	ClientId    string          `json:"client_id"`
	Amount      float64         `json:"amount"`
	PaymentType string          `json:"payment_type"`
	PaymentDate string          `json:"payment_date"`
	PaymentInfo CardPaymentInfo `json:"payment_info"`
}

type CardPaymentInfo struct {
	MaskedNumber string `json:"masked_number"`
	CardToken    string `json:"-"`
}

type Service interface {
	CreatePayment(payment *CardPayment) error
}
